modules/gapi/src/executor/gstreamingexecutor.cpp

```diff
@@ -553,6 +555,98 @@ void cv::gimpl::GStreamingExecutor::setSource(GRunArgs &&ins)
         util::throw_error(std::logic_error("Only one video source is"
                                            " currently supported!"));
     }
+ 
+    // testing meta data is cv::GMatDesc{CV_8U,3,cv::Size{768,576}}
+    if (m_metas.empty()) {
+        // extract meta from first frame
+    } else {
+        //const auto& metas = m_metas;
+    }
+    
+    const auto& metas = m_metas;
+    
+    if (/*wasFinished*/true) {
+        auto pass_ctx = ade::passes::PassContext{*m_orig_graph.get()};
+        cv::gimpl::passes::initMeta(pass_ctx, metas);
+        //does inferMeta needed?? 
+        cv::gimpl::passes::inferMeta(pass_ctx, true);
+        //compile islands for m_orig_graph
+        cv::gimpl::passes::storeResultingMeta(pass_ctx);
+        // Get compileArgs from m_ops?? 
+        cv::gimpl::GCompiler::compileIslands(*m_orig_graph.get(), comp_args);
```

Comment:
Please move this partial compilation code somewhere to `gcompiler.cpp`, maybe making it a static method of `GCompiled`. We don't need _this_ level of details here, it should be kept closer to the general compiler logic. This `GStreamingExecutor` should be just an _user_ of that logic.

Reply:
Now its a static method of `GCompiled`
`cv::GCompiled::setMetaData(ade::Graph &g, cv::GCompileArgs &args, const cv::GMetaArgs &metas)`